tc/sgx/trusted_worker_manager/enclave_untrusted/enclave_bridge/signup.cpp

```diff
@@ -192,7 +190,218 @@ tcf_err_t tcf::enclave_api::enclave_data::CreateEnclaveData(
+            g_Enclave[0].CallSgx(
+                [&target_info,
+                 &epidGroupId] () {
+                    return sgx_init_quote(&target_info, &epidGroupId);
+                });
+        tcf::error::ThrowSgxError(sresult,
+            "Intel SGX enclave call failed (sgx_init_quote);"
+            " failed to initialize the quote");
+
+        // Properly size the sealed signup data buffer for the caller
+        // and call into the enclave to create the signup data
+        sgx_report_t enclave_report = { 0 };
+
+        sresult = g_Enclave[0].CallSgx(
+            [enclaveid,
+             &presult,
+             target_info,
+             inExtData,
+             inExtDataSignature,
+             &outPublicEnclaveData,
+             &sealed_enclave_data_buffer,
+             &enclave_report ] () {
+                sgx_status_t ret = ecall_CreateSignupDataKME(
+                    enclaveid,
+                    &presult,
+                    &target_info,
+                    (const uint8_t*) inExtData.c_str(),
+                    inExtData.length(),
+                    (const uint8_t*) inExtDataSignature.c_str(),
+                    inExtDataSignature.length(),
+                    outPublicEnclaveData.data(),
+                    outPublicEnclaveData.size(),
+                    sealed_enclave_data_buffer.data(),
+                    sealed_enclave_data_buffer.size(),
+                    &enclave_report);
+                return tcf::error::ConvertErrorStatus(ret, presult);
+            });
+        tcf::error::ThrowSgxError(sresult,
+            "Intel SGX enclave call failed (ecall_CreateSignupData);"
+            " failed to create signup data");
+        g_Enclave[0].ThrowTCFError(presult);
+        outSealedEnclaveData = \
+            ByteArrayToBase64EncodedString(sealed_enclave_data_buffer);
+
+        // Take the report generated and create a quote for it, encode it
+        size_t quote_size = tcf::enclave_api::base::GetEnclaveQuoteSize();
+        ByteArray enclave_quote_buffer(quote_size);
+        g_Enclave[0].CreateQuoteFromReport(&enclave_report, enclave_quote_buffer);
+        outEnclaveQuote = ByteArrayToBase64EncodedString(enclave_quote_buffer);
+    } catch (tcf::error::Error& e) {
+        tcf::enclave_api::base::SetLastError(e.what());
+        result = e.error_code();
+    } catch (std::exception& e) {
+        tcf::enclave_api::base::SetLastError(e.what());
+        result = TCF_ERR_UNKNOWN;
+    } catch (...) {
+        tcf::enclave_api::base::SetLastError(
+            "Unexpected exception in (CreateEnclaveData)");
+        result = TCF_ERR_UNKNOWN;
+    }
+
+    return result;
+}  // tcf::enclave_api::base::CreateEnclaveDataKME
+
+// XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX
+tcf_err_t tcf::enclave_api::enclave_data::CreateEnclaveDataWPE(
+    const std::string& inExtData,
+    const std::string& inExtDataSignature,
+    const std::string& inKmeAttestaion,
```

Comment:
Change inKmeAttestaion to inKmeAttestation here and everywhere else below.

Reply:
Done